### ISSUE TITLE
[clang-tidy] readability-redundant-preprocessor false positive on builtin preprocessor checks

### ISSUE THREAD

**Issue by philnik777** [llvm/llvm-project#64825]
```c++
#if __has_builtin(__remove_cvref)
#  if __has_cpp_attribute(no_unique_address)
#  endif
#endif
```
([Godbolt](https://godbolt.org/z/sEs3P3Yz6)) generates
```c++
1 warning generated.
<source>:2:4: warning: nested redundant #if; consider removing it [readability-redundant-preprocessor]
    2 | #  if __has_cpp_attribute(no_unique_address)
      |    ^
<source>:1:2: note: previous #if was here
    1 | #if __has_builtin(__remove_cvref)
      |  ^
1 warning generated.
```
which is obviously wrong.

**Comment by PiotrZSL** [comment]
What I found is that ConditionRange point to ')' character only, and we compare later ')' with ')', oryginal issue is probably somewhere in clang preprocessor, but still easiest fix would be simply to take as a begining of range a token after 'if'.

**Comment by Aditya26189** [comment]
Following up on @PiotrZSL's finding - it looks like `ConditionRange` is pointing to just the `)` character, so different conditions like `__has_builtin(__remove_cvref)` and `__has_cpp_attribute(no_unique_address)` both appear as just `)`.

What if we use `getCharRange()` instead of `getTokenRange()` when extracting the condition text? That might capture the full expression.

I will be happy to work on a fix and add test cases if this approach sounds reasonable. Let me know if I can help!
